tests/unit/test_expression_dependent_columns.py

```diff
@@ -0,0 +1,29 @@
+#
+# Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+#
+
+from snowflake.snowpark._internal.analyzer.binary_expression import Add
+from snowflake.snowpark._internal.analyzer.expression import UnresolvedAttribute
+from snowflake.snowpark._internal.analyzer.grouping_set import GroupingSetsExpression
+from snowflake.snowpark._internal.analyzer.unary_expression import Alias
+
+
```

Comment:
As we discussed offline, let's have a jira and TODO here for adding more unit tests